### DISCUSSION TITLE
How do I send a command to the last active window of another os window?

### DISCUSSION THREAD

**Discussion by timgsl** [kovidgoyal/kitty#5647] [Q&A] [answered]
<img width="1569" alt="image" src="https://user-images.githubusercontent.com/99874400/200221399-35d7a86a-823c-4ed0-a8f4-afe21c595aad.png">

I have 2 kitty os windows open. I would like to send a command to the last active window of the inactive os window.

Is there any way using kitty @ send-text --match  to match the last active window of the inactive os window ?

Failing which, I suppose I have to create a kitten and use the info below in kitty @ ls to send the command
- os_window.is_focused
- os_window.tabs.active_window_history

I found a similar discussion here https://github.com/kovidgoyal/kitty/discussions/5021 but I was wondering if there would be a more convenient method via kitty @ send-text --match ...

**Comment by timgsl** [comment]
Ah this command will work well enough`kitty @ send-text --match 'not state:focused and not state:parent_focused'`
It sends the text to all the tabs in any unfocussed os windows but that's still workable for me ...

**Comment by kovidgoyal** [comment, marked as answer]
state:active and not state:focused

but you should run from nightly as there were some fixes with regards to
this recently.
